builder/builder.go

```diff
@@ -1952,44 +1969,74 @@ func createDeltaPacks(from *swupd.Manifest, to *swupd.Manifest, printReport bool
 	sort.Strings(orderedBundles)
 
 	for _, name := range orderedBundles {
-		bp := bundlesToPack[name]
-		packPath := filepath.Join(outputDir, fmt.Sprint(bp.ToVersion), swupd.GetPackFilename(bp.Name, bp.FromVersion))
+		b := bundlesToPack[name]
+		packPath := filepath.Join(outputDir, fmt.Sprint(b.ToVersion), swupd.GetPackFilename(b.Name, b.FromVersion))
 		_, err = os.Lstat(packPath)
 		if err == nil {
-			fmt.Printf("  Delta pack already exists for %s from %d to %d\n", bp.Name, bp.FromVersion, bp.ToVersion)
+			fmt.Printf("  Delta pack already exists for %s from %d to %d\n", b.Name, b.FromVersion, b.ToVersion)
+			// Remove so the goroutines don't try to make deltas for these
+			delete(bundlesToPack, name)
 			continue
 		}
 		if !os.IsNotExist(err) {
 			return errors.Wrapf(err, "couldn't access existing pack file %s", packPath)
 		}
-		fmt.Printf("  Creating delta pack for bundle %q from %d to %d\n", bp.Name, bp.FromVersion, bp.ToVersion)
-		info, err := swupd.CreatePack(bp.Name, bp.FromVersion, bp.ToVersion, outputDir, bundleDir, numWorkers)
-		if err != nil {
-			return err
-		}
+	}
+	// Cap the routines for bundles to the number of bundles
+	var bundleWorkers int
+	if numWorkers < 1 {
+		numWorkers = 1
+		bundleWorkers = 1
+	} else {
+		bundleWorkers = runtime.NumCPU()
```

Comment:
so if numWorkers == 5 you still set bundleWorkers to NumCPU?

Reply:
Yes, ty for pointing it out, I actually want to cap at NumCPU() always unless 0 is passed in specifically, since it's no longer creating deltas as part of the tar pack creation.